internal/signprovider/ecdsa/x509/provider.go

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright 2024 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package x509
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/config"
+	"math/big"
+	"os"
+)
+
+// provider is a receiver that encapsulates required dependencies.
+type provider struct{}
+
+// New is a factory function that returns an initialized provider.
+func New() *provider {
+	return &provider{}
+}
+
+func (p *provider) Sign(key config.KeyInfo, content []byte) (string, error) {
+	prvKeyBytes, err := os.ReadFile(key.Path)
+	if err != nil {
+		return "", err
+	}
+
+	blockPriv, _ := pem.Decode(prvKeyBytes)
+
+	keyDecoded, err := x509.ParseECPrivateKey(blockPriv.Bytes)
+	if err != nil {
+		return "", err
+	}
+
+	r, s, err := ecdsa.Sign(rand.Reader, keyDecoded, content)
+	if err != nil {
+		return "", err
+	}
+	return string(append(r.Bytes(), s.Bytes()...)), nil
+
+}
+
+func (p *provider) Verify(key config.KeyInfo, content, signature []byte) (bool, error) {
+	pubKeyBytes, err := os.ReadFile(key.Path)
+	if err != nil {
+		return false, err
+	}
+
+	blockPub, _ := pem.Decode(pubKeyBytes)
+
+	keyDecoded, err := x509.ParsePKIXPublicKey(blockPub.Bytes)
+	if err != nil {
+		return false, err
+	}
+
+	r := new(big.Int).SetBytes(signature[:len(signature)/2])
+	s := new(big.Int).SetBytes(signature[len(signature)/2:])
+	return ecdsa.Verify(keyDecoded.(*ecdsa.PublicKey), content, r, s), nil
```

Comment:
Similarly, `Verify` [expects a hash](https://pkg.go.dev/crypto/ecdsa#Verify) rather than the actual content.

Reply:
Fixed